pykafka/producer.py

```diff
@@ -140,6 +141,14 @@ def __init__(self,
         self._update_lock = self._cluster.handler.Lock()
         self.start()
 
+        def cleanup(obj):
+            if obj._running:
+                obj.stop()
```

Comment:
Would it work? 
<pre>
def get_kafka_async_producer(topic_name):
    """
    Return a real kafka producer for dev, stage and prod. For unit-tests it returns None
    """
    try:
        with KafkaProducer(topic_name, async=True, async_queue_maxsize=settings.KAFKA_ASYNC_QUEUE_MAXSIZE) as producer:
            return producer
    except KafkaUnavailableError:
        log.exception("Connection to Kafka is unavailable")
        return None
</pre>

In the ``KafkaProducer()`` class we instantiate the pykafka ``SimpleProducer()`` class.